Descent3/init.cpp

```diff
@@ -1690,38 +1690,20 @@ void InitIOSystems(bool editor) {
 #endif
 
   // Open this file if it's present for stuff we might add later
-  ddio_MakePath(fullname, LocalD3Dir, "extra.hog", NULL);
-  extra_hid = cf_OpenLibrary(fullname);
-
-  // Opens the mercenary hog if it exists and the registry entery is present
-  // Win32 only.  Mac and Linux users are SOL for now
-#ifdef WIN32
-  HKEY key;
-  DWORD lType;
-  LONG error;
-
-#define BUFLEN 2000
-  char dir[BUFLEN];
-  DWORD dir_len = BUFLEN;
-
-  error =
-      RegOpenKeyEx(HKEY_LOCAL_MACHINE, "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\Descent3 Mercenary", 0,
-                   KEY_ALL_ACCESS, &key);
-
-  if ((error == ERROR_SUCCESS)) {
-    lType = REG_EXPAND_SZ;
-
-    unsigned long len = BUFLEN;
-    error = RegQueryValueEx(key, "UninstallString", NULL, &lType, (unsigned char *)dir, &len);
+  // JC: Digital releases call this 'extra1'hog', on CD releases it's 'extra.hog'. Check for extra1.hog first.
+  //ddio_MakePath(fullname, LocalD3Dir, "extra1.hog", NULL);
+  extra_hid = cf_OpenLibrary("extra1.hog");
+  if (extra_hid == -1) {
+	//ddio_MakePath(fullname, LocalD3Dir, "extra.hog", NULL);
+ 	extra_hid = cf_OpenLibrary("extra.hog");
+  }
 
-    if (error == ERROR_SUCCESS) {
-      merc_hid = cf_OpenLibrary("merc.hog");
-    }
+  // Always look for extra.hog first
+  merc_hid = cf_OpenLibrary("extra.hog");
+  // Check if merc_hid is valid, if not, open merc.hog
+  if (merc_hid == -1) {
+    merc_hid = cf_OpenLibrary("merc.hog");
```

Comment:
I want to make sure that I’m understanding this code correctly. Would this make it so that `merc.hog` only gets loaded if `extra.hog` fails to load?

Reply:
Correct. If you have `extra1.hog`, then the mercenary assets are in `extra.hog`. If you have `extra.hog`, mercenary assets are in `merc.hog`.

Reply:
I installed the Windows version Descent 3+Mercenary from CDs, and I have both `extra.hog` and `merc.hog`. In that scenario, I think that this code would make it so that `merc.hog` doesn’t get loaded. This code would load `extra.hog` successfully and then ignore `merc.hog`.